models/remoteworkitem.go

```diff
@@ -0,0 +1,96 @@
+package models
+
+import (
+	"fmt"
+	"strconv"
+)
+
+const (
+	ProviderGithub string = "github"
+)
+
+// RemoteWorkItem represents a work item as it is stored in the database
+type RemoteWorkItem struct {
+	Fields map[string]interface{}
+}
+
+// WorkItemMap : holds the mapping between Remote key and local key in a WI
+type WorkItemMap map[string]interface{}
+
+// RemoteWorkItemSanitizer makes sure that the
+// Response from Provider API is sanitized into a
+// good looking flat 1-level dictionary
+type RemoteWorkItemMapper interface {
+	Flatten()
```

Comment:
The other alternative would be to **not** have derivatives of `RemoteWorkItem` . The `MapRemote` method could conditionally decide how to flatten based on provider.


// MapRemote maps RemoteWorkItem to WorkItem
func (rwi *RemoteWorkItem) MapRemote(wiMap WorkItemMap, wiType *WorkItemType , Provider string ) WorkItem {
    //  Based on Provider call appropriate flattening method.
}
```